src/Nethermind/Nethermind.JsonRpc/Modules/RpcModuleProvider.cs

```diff
@@ -156,21 +211,67 @@ private class ResolvedMethodInfo
             {
                 ModuleType = moduleType;
                 MethodInfo = methodInfo;
-                ExpectedParameters = methodInfo.GetParameters();
+
+                ParameterInfo[] parameters = methodInfo.GetParameters();
+                ExpectedParameter[] expectedParameters = new ExpectedParameter[parameters.Length];
+                for (var i = 0; i < parameters.Length; i++)
+                {
+                    ParameterInfo parameter = parameters[i];
+                    ParameterDetails details = ParameterDetails.None;
+                    if (parameter.ParameterType.IsAssignableTo(typeof(IJsonRpcParam)))
+                    {
+                        details |= ParameterDetails.IsIJsonRpcParam;
+                    }
+                    if (IsNullableParameter(parameter))
+                    {
+                        details |= ParameterDetails.IsNullable;
+                    }
+                    if (parameter.IsOptional)
+                    {
+                        details |= ParameterDetails.IsOptional;
+                    }
+
+                    expectedParameters[i] = new(parameter, details);
+                }
+
+                ExpectedParameters = expectedParameters;
                 ReadOnly = readOnly;
                 Availability = availability;
+                Invoker = MethodInvoker.Create(methodInfo);
             }
 
             public string ModuleType { get; }
             public MethodInfo MethodInfo { get; }
-            public ParameterInfo[] ExpectedParameters { get; }
+            public MethodInvoker Invoker { get; }
+            public ExpectedParameter[] ExpectedParameters { get; }
             public bool ReadOnly { get; }
             public RpcEndpoint Availability { get; }
 
             public override string ToString()
             {
                 return MethodInfo.Name;
             }
+
+            private static bool IsNullableParameter(ParameterInfo parameterInfo)
+            {
+                Type parameterType = parameterInfo.ParameterType;
+                if (parameterType.IsValueType)
+                {
+                    return Nullable.GetUnderlyingType(parameterType) is not null;
+                }
+
+                CustomAttributeData nullableAttribute = parameterInfo.CustomAttributes
+                    .FirstOrDefault(x => x.AttributeType.FullName == "System.Runtime.CompilerServices.NullableAttribute");
```

Comment:
is by string check fastest here? Not like type id?

Reply:
I just move the code; with this change its only called once on startup and then caches the bool, so not sure it matters too much?

Perviously it would call this for every call and every parameter

Reply:
Updated to use generic lookup rather than Linq and string compares